subsys/modem/modem_chat.c

```diff
@@ -835,3 +836,104 @@ void modem_chat_release(struct modem_chat *chat)
 	chat->matches[MODEM_CHAT_MATCHES_INDEX_RESPONSE] = NULL;
 	chat->matches_size[MODEM_CHAT_MATCHES_INDEX_RESPONSE] = 0;
 }
+
+void modem_chat_match_runtime_init(struct modem_chat_match_runtime *runtime, uint8_t *match_buf,
+				   uint8_t match_buf_size, uint8_t *separators_buf,
+				   uint8_t separators_buf_size)
+{
+	__ASSERT(runtime != NULL, "runtime must be provided");
+	__ASSERT(match_buf != NULL, "match_buf must be provided");
+	__ASSERT(match_buf_size > 1, "match_buf_size must more than ");
+	__ASSERT(separators_buf != NULL, "separators_buf must be provided");
+	__ASSERT(separators_buf_size > 0, "separators_buf_size must more than 0");
```

Comment:
Will these only assert if assert is enabled (kconfig) or will they always assert?

Reply:
Only if enabled, the idea being that you would disable the asserts when building production code to save space/overhead :)

Reply:
Got it. Wouldn't we want this stuff checked without asserts?

Reply:
I only place these asserts in init functions which are only run once, usually in the device drivers init function.This is to fail fast since there is no recovery action that can be made if the init fails.
This pattern turns this
```
int ret = modem_chat_match_init(&foo, &bar);
if (ret < 0) {
        return ret;
}
```
into the simpler
```
modem_chat_match_init(&foo, &bar);
```
which simply explodes if not initialized properly. Then, for production code, the asserts are disabled, which is safe as the variables passed to the init functions don't change at runtime, they are statically allocated buffers and const configurations :)